### PR TITLE
Added support for the Watchy board by SQFMI.

### PR DESCRIPTION
**Pull request by joshua-beck-0908**
I've been using a SQFMI Watchy board for a while so I thought I might add CircuitPython support to it.
Website: https://watchy.sqfmi.com/

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -383,3 +383,12 @@
...
 	url = https://github.com/adafruit/Adafruit_CircuitPython_SHT4x
+[submodule "frozen/Adafruit_CircuitPython_NTP"]
+	path = frozen/Adafruit_CircuitPython_NTP
+	url = https://github.com/adafruit/Adafruit_CircuitPython_NTP.git
+[submodule "frozen/Adafruit_CircuitPython_PCF8563"]
+	path = frozen/Adafruit_CircuitPython_PCF8563
+	url = https://github.com/adafruit/Adafruit_CircuitPython_PCF8563.git
+[submodule "frozen/CircuitPython_BMA423"]
+	path = frozen/CircuitPython_BMA423
+	url = https://github.com/jposada202020/CircuitPython_BMA423.git
```
Comment (tannewt) [inline .gitmodules:394]: Please don't add more frozen libraries unless you need them to free up RAM. Libraries outside of the image are much easier to update.

File: ports/espressif/boards/sqfmi_watchy/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+CIRCUITPY_ESP_FLASH_FREQ=40m
+CIRCUITPY_ESP_FLASH_SIZE=4MB
+
+IDF_TARGET = esp32
+
+#FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_PCF8563
+#FROZEN_MPY_DIRS += $(TOP)/frozen/CircuitPython_BMA423
+#FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_BusDevice
+#FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_Register
+#FROZEN_MPY_DIRS += $(TOP)/frozen/Adafruit_CircuitPython_NTP
```
Comment (tannewt) [inline ports/espressif/boards/sqfmi_watchy/mpconfigboard.mk:21]:
```suggestion
```

File: ports/espressif/boards/sqfmi_watchy/mpconfigboard.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
+CIRCUITPY_CREATOR_ID = 0x4496E3F4
+CIRCUITPY_CREATION_ID = 0x00320024
+
+USB_VID = 0x10C4
+USB_PID = 0xEA60
+USB_PRODUCT = "Watchy"
+USB_MANUFACTURER = "SQFMI"
```
Comment (tannewt) [inline ports/espressif/boards/sqfmi_watchy/mpconfigboard.mk:7]:
```suggestion
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for adding this! Just a few comments to address.

File: .gitmodules (hunk 1 above)
Comment (joshua-beck-0908) [reply .gitmodules:394]: Saving RAM is the intent of freezing these libraries, I've only frozen fairly stable libraries required by the hardware; but if you require me to take out the frozen libraries I will.

Comment (tannewt) [reply .gitmodules:394]: Are you saving ram preemptively or did you have demo code already need this? I'd rather only add it when it is needed.

Comment (joshua-beck-0908) [reply .gitmodules:394]: Mostly from practical testing. I have had this firmware for some time and just decided to update it to CP 9 and commit.
Originally I didn't freeze anything, but by the time I added all the hardware libraries, ePaper display driver, graphics libraries and a bitmap for the watch face it was starting to run low on memory. The perils of ESP32 without PSRAM I suppose. I'll let you be the judge on whether this is sufficient justification to add frozen libraries.

Comment (tannewt) [reply .gitmodules]: Ya, I think that's enough reason.

**Review by tannewt — APPROVED** [review]
Looks good. Will wait to hear if you want to freeze more.

**Comment by tannewt** [comment]
Merging. Feel free to open a new PR to freeze things in.

**Pull request merged**